### PR TITLE
Intly 6733

### PR DESCRIPTION
**Pull request by philbrookes**
# Description
Allowing customers to configure a schedule for upgrades to be applied

@davidffrench 

## Type of change
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Verified independently on a cluster by reviewer

### CODE REVIEW

File: pkg/controller/subscription/subscription_controller.go [hunk 1]
```diff
@@ -73,13 +83,74 @@ func (r *ReconcileSubscription) Reconcile(request reconcile.Request) (reconcile.
...
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: 10 * time.Second,
+		}, nil
+	}
+	logrus.Infof("not automatically upgrading a Service Affecting Release")
+	return reconcile.Result{
+		Requeue:      true,
+		RequeueAfter: 10 * time.Second,
+	}, nil
```
Comment (davidffrench) [inline pkg/controller/subscription/subscription_controller.go:155]: Should this requeue every 10 seconds when we don't want to upgrade yet? Would it be better to increase the time?

File: Makefile [hunk 2]
```diff
@@ -73,6 +73,10 @@ setup/git/hooks:
 code/run: code/gen cluster/prepare/smtp cluster/prepare/dms cluster/prepare/pagerduty
 	@$(OPERATOR_SDK) run --local --namespace="$(NAMESPACE)"
 
+.PHONY: code/rerun
+code/rerun:
+	@$(OPERATOR_SDK) run --local --namespace="$(NAMESPACE)"
+
```
[the remark this answers is not in the document]
Comment (philbrookes) [reply Makefile:79]: yeah, code/run was taking an annoyingly long time. I didn't want to remove the pre-steps in code/run cos I like them, but stop/starting frequently was frustrating.

File: deploy/crds/examples/integreatly-rhmi-config-cr.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
 apiVersion: integreatly.org/v1alpha1
 kind: RHMIConfig
 metadata:
-  name: rhmi-configuration
\ No newline at end of file
+  name: cloud-resource-config
```
[the remark this answers is not in the document]
Comment (philbrookes) [reply deploy/crds/examples/integreatly-rhmi-config-cr.yml:4]: to match what's hard-coded in the codebase.

File: pkg/controller/subscription/subscription_controller.go [hunk 4]
```diff
@@ -38,7 +46,7 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 		return err
 	}
 
-	err = c.Watch(&source.Kind{Type: &v1alpha1.Subscription{}}, &handler.EnqueueRequestForObject{})
+	err = c.Watch(&source.Kind{Type: &operatorsv1alpha1.Subscription{}}, &handler.EnqueueRequestForObject{})
```
[the remark this answers is not in the document]
Comment (philbrookes) [reply pkg/controller/subscription/subscription_controller.go:49]: yeah, we filter by name in the reconciler

File: pkg/controller/subscription/subscription_controller.go (hunk 1 above)
Comment (philbrookes) [reply pkg/controller/subscription/subscription_controller.go]: good point, updated to once a minute

File: pkg/controller/subscription/subscription_controller.go [hunk 5]
```diff
@@ -73,13 +83,74 @@ func (r *ReconcileSubscription) Reconcile(request reconcile.Request) (reconcile.
...
+	}
+
+	latestRHMICSV, err := rhmiConfigs.GetCSV(latestRHMIInstallPlan)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	config := &integreatlyv1alpha1.RHMIConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      installation.DefaultCloudResourceConfigName,
```
Comment (KevFan) [inline pkg/controller/subscription/subscription_controller.go:119]: Should this not be `rhmi-config`? :thinking: 
https://github.com/integr8ly/integreatly-operator/blob/8b6d1aa5157a7135bd7353a7f99814fff15db2b4/pkg/controller/installation/bootstrapReconciler.go#L129

`cloud-resource-config` is the name of the cloud resource config map for forcing whether to use in cluster or aws storage I believe

File: deploy/crds/examples/integreatly-rhmi-config-cr.yml (hunk 3 above)
Comment (KevFan) [inline deploy/crds/examples/integreatly-rhmi-config-cr.yml:4]: Same, I think this should be `rhmi-config`

File: pkg/controller/subscription/subscription_controller.go (hunk 5 above)
Comment (philbrookes) [reply pkg/controller/subscription/subscription_controller.go:119]: oh, this is strange, is `DefaultCloudResourceConfigName` used anywhere then?

Comment (philbrookes) [reply pkg/controller/subscription/subscription_controller.go]: oh, it's the name of the configmap for CRO :facepalm: I'll update now

**Comment by philbrookes** [comment]
/retest
flake logged

**Pull request merged**